code/modules/mob/dead/observer/observer.dm

```diff
@@ -495,7 +495,7 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	var/mob/M = get_top_transmogrification()
 	if(!M.client)
 		return
-	if(!(mind && mind.current && can_reenter_corpse))
+	if(!(mind && mind.current && can_reenter_corpse) || M == src)
```

Comment:
What's this change for? Because it seems to make it impossible to reenter your corpse.